lib/posix/options/syslog.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/posix/syslog.h>
+
+static struct k_spinlock syslog_lock;
+static uint8_t syslog_mask;
+
+static int syslog_priority_to_zephyr_log_level(int priority)
+{
+	switch (priority) {
+	case LOG_EMERG:
+	case LOG_ALERT:
+	case LOG_CRIT:
+	case LOG_ERR:
+		return LOG_LEVEL_ERR;
+	case LOG_WARNING:
+		return LOG_LEVEL_WRN;
+	case LOG_NOTICE:
+	case LOG_INFO:
+		return LOG_LEVEL_INF;
+	case LOG_DEBUG:
+		return LOG_LEVEL_DBG;
+	default:
+		return -EINVAL;
+	}
+}
+
+void closelog(void)
+{
+}
+
+void openlog(const char *ident, int option, int facility)
+{
+	ARG_UNUSED(ident);
+	ARG_UNUSED(option);
+	ARG_UNUSED(facility);
+}
+
+void syslog(int priority, const char *format, ...)
+{
+	va_list ap;
+
+	va_start(ap, format);
+	vsyslog(priority, format, ap);
+	va_end(ap);
+}
+
+int setlogmask(int maskpri)
+{
+	int oldpri;
+
+	K_SPINLOCK(&syslog_lock) {
+		oldpri = syslog_mask;
+		syslog_mask = maskpri;
+	}
+
+	return oldpri;
+}
+
+void vsyslog(int priority, const char *format, va_list ap)
+{
+	uint8_t mask;
+	int level = syslog_priority_to_zephyr_log_level(priority);
+
+	if (level < 0) {
+		/* invalid priority */
+		return;
+	}
+
+	K_SPINLOCK(&syslog_lock) {
+		mask = syslog_mask;
+	}
+
+	if ((BIT(level) & mask) == 0) {
+		/* masked */
+		return;
+	}
+
+	log2_generic(level, format, ap);
```

Comment:
@nordic-krch - is there a particular Kconfig option I should check for here? If the `LOG_CONS` option is specified and logging is expected to be a no-op, I should probably use `printk()` instead.

Reply:
I think this could do the trick. Let me know if it works for you.

Reply:
don't you want to add level when minimal logging is used?

I thought i removed all remains of v2 naming in logging but somehow that was left.

Reply:
@nordic-krch - I hope it's ok, I prepended a commit to deprecate `log2_generic()` and added `log_generic()` (basically just a rename).

It's possible that log level, time, and other items could be added in the future, but I would prefer to avoid multiple calls to `(v)printk()` at this point because it's possible that messages might not be printed atomically.

Reply:
>  I prepended a commit to deprecate log2_generic()

Thanks for doing that!